File: 数组/双指针/344. 反转字符串.js
// 属于左右指针题目

var reverseString = function (s) {
    let left = 0;
    let right = s.length - 1;
    while (left < right) {
        [s[left], s[right]] = [s[right], s[left]];
        left++;
        right--;
    }
    console.log(s);
};

reverseString(["h", "e", "l", "l", "o"]);

File: 双指针/相向双指针/344. 反转字符串.py
class Solution(object):
    def reverseString(self, s):      
        left = 0
        right = len(s) - 1
        
        while left < right:
            s[left], s[right] = s[right] ,s[left]
            left += 1
            right -= 1
        return s

Solution().reverseString(s = ["h","e","l","l","o"])
